src/main/java/edu/cornell/kfs/coa/businessobject/CuAccountGlobal.java

```diff
@@ -2,164 +2,903 @@
+		}
+	}
     
-    public String getMajorReportingCategoryCode() {
+	private Account updateAccountValuesOnAccountGlobalDetailWithNewValuesFromAccountGlobalDoc(
+			GlobalBusinessObjectDetailBase globalDetail) {
+		AccountGlobalDetail accountGlobalDetail = (AccountGlobalDetail) globalDetail;
+		Account account = getBusinessObjectService().findByPrimaryKey(Account.class,accountGlobalDetail.getPrimaryKeys());
+
+		if (ObjectUtils.isNotNull(account)) {
+			updateAccountBasicFields(account);
+			updateAccountExtendedAttribute(account);
+			
+			if(ObjectUtils.isNotNull(this.getIndirectCostRecoveryAccounts()) && this.getIndirectCostRecoveryAccounts().size() > 0){
+				updateIcrAccounts(globalDetail, account.getIndirectCostRecoveryAccounts());
+			}
+		}
+
+		return account;
+	}
+
+	/**
+	 * The business rules for the accountRestrictedStatusCode is as follows:
+	 *
+	 * If the sub-fund of the account listed in the edit list of accounts on
+	 * the global maintenance edoc is one that has a default accountRestrictedStatusCode
+	 * listed, then any change entered in the global account maintenance edoc is ignored.
+	 *
+	 * If the sub-fund of the account listed in the edit list of accounts on
+	 * the global maintenance edoc is one that does NOT have a default accountRestrictedStatusCode
+	 * listed, then any change entered in the global account maintenance edoc overwrites
+	 * the accountRestrictedStatusCode value on that particular account from the edit list.
+	 *
+	 * @param globalAccountMaintenanceDocSubFundGroup User entered Sub-Fund Group for the Sub-Fund Group Code.
+	 * @param globalAccountMaintenanceDocRestrictedStatusCode User entered Account Restricted Status Code value.
+	 * @param accountBeingEdited The Single Account to apply the business rules to.
+	 */
+	private void updateAccountRestrictedStatusCodeForAccountBeingEdited(SubFundGroup globalAccountMaintenanceDocSubFundGroup, String globalAccountMaintenanceDocRestrictedStatusCode, Account accountBeingEdited)
+	{
+	    if (ObjectUtils.isNull(globalAccountMaintenanceDocSubFundGroup)) {
+	        if ( !isDefaultAccountRestrictedStatusCodeSetOnSubFundGroup(accountBeingEdited.getSubFundGroup()) ) {
+	            accountBeingEdited.setAccountRestrictedStatusCode(globalAccountMaintenanceDocRestrictedStatusCode);
+	        }
+	    }
+	    else {
+	        if ( !isDefaultAccountRestrictedStatusCodeSetOnSubFundGroup(globalAccountMaintenanceDocSubFundGroup) ) {
+	            accountBeingEdited.setAccountRestrictedStatusCode(globalAccountMaintenanceDocRestrictedStatusCode);
+	        }
+	    }
+	}
+
+	/**
+	 *
+	 * @param subFundGroup
+	 * @return true when a default account restricted status code set on the sub-fund; false otherwise
+	 */
+	private boolean isDefaultAccountRestrictedStatusCodeSetOnSubFundGroup(SubFundGroup subFundGroup) {
+		if (ObjectUtils.isNotNull(subFundGroup) && StringUtils.isNotBlank(subFundGroup.getAccountRestrictedStatusCode())) {
+			return true;
+		}
+		return false;
```

Comment:
It would probably be simpler and cleaner to just return the result of the condition instead.